examples/forbid-anonymous-operations/src/forbid_anonymous_operations.rs

```diff
@@ -0,0 +1,235 @@
+use apollo_router_core::{
+    checkpoint::Step, plugin_utils, register_plugin, Plugin, RouterRequest, RouterResponse,
+    ServiceBuilderExt,
+};
+use http::StatusCode;
+use tower::{util::BoxService, BoxError, ServiceBuilder, ServiceExt};
+
+#[derive(Default)]
+// Global state for our plugin would live here.
+// We don't need any in this example
+struct ForbidAnonymousOperations {}
+
+impl Plugin for ForbidAnonymousOperations {
+    // We either forbid anonymous operations,
+    // Or we don't. This is the reason why we don't need
+    // to deserialize any configuration from a .yml file.
+    //
+    // Config is a unit, and `ForbidAnonymousOperation` derives default.
+    type Config = ();
+
+    fn new(_configuration: Self::Config) -> Result<Self, BoxError> {
+        Ok(Self::default())
+    }
+
+    // Forbidding anonymous operations can happen at the very beginning of our GraphQL request lifecycle.
+    // We will thus put the logic it in the `router_service` section of our plugin.
+    fn router_service(
+        &mut self,
+        service: BoxService<RouterRequest, RouterResponse, BoxError>,
+    ) -> BoxService<RouterRequest, RouterResponse, BoxError> {
+        // `ServiceBuilder` provides us with a `with_checkpoint` method.
+        //
+        // This method allows us to return Step::Continue(request) if we want to let the request through,
+        // or Step::Return(response) with a crafted response if we don't want the request to go through.
+        ServiceBuilder::new()
+            .with_checkpoint(|req: RouterRequest| {
```

Comment:
`with_checkpoint` should be renamed to `checkpoint`, but maybe do that as part of the async checkpoint PR.

Reply:
so it's not, let me rename it!